Add secondary license to LICENSE file

ndoschek opened the pull request:
- Align all licenses to use MIT as secondary license
- As this project was initially based on other projects, the usage of the GPL license were copy paste mistakes. The initial setup of the project forsaw to use MIT as secondary license

ndoschek committed:
Add secondary license to LICENSE files

- Align all licenses to use MIT as secondary license
- As this project was initially based on other projects, the usage of the GPL license were copy paste mistakes. The initial setup of the project forsaw to use MIT as secondary license

lucas-koehler commented on client/packages/theia-ecore/LICENSE:
```diff
@@ -54,8 +59,7 @@ in any manner that enables the transfer of a copy.
 modifications, including but not limited to software source code,
 documentation source, and configuration files.
 
-"Secondary License" means either the GNU General Public License,
```
Is this correct? It was GPL before and now is MIT?

lucas-koehler commented on client/packages/sprotty-ecore/LICENSE:
```diff
@@ -54,8 +59,7 @@ in any manner that enables the transfer of a copy.
 
-"Secondary License" means either the GNU General Public License,
-Version 2.0, or any later versions of that license, including any
+"Secondary License" means either the MIT License or any
```
Is this correct? It was GNU before and now is MIT?

ndoschek commented:
Thanks for catching that, I totally overlooked that! Hope it is correct now.

ndoschek committed:
Update Jenkins build

- Use eclipseglsp/ci:uitest-v1.1 as the project requires Node 12
- Use github token to stabilize client client

ndoschek commented:
After doublechecking with Philip and Eugen, we agreed to use MIT as secondary license, as the mixup with GPL was a mistake on project setup, it was intended to use MIT from the start.
Also, I updated the docker image to fix the Jenkins build and ensure the required node 12 is used.

eneufeld reviewed (approved):
Thank you!
